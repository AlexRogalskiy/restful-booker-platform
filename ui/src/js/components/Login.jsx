import React from 'react';
import Cookies from 'universal-cookie';

export default class Login extends React.Component {

    constructor() {
        super();

        this.doLogin = this.doLogin.bind(this);

        this.state = {
            error : false
        }
    }

    doLogin() {
        fetch('http://' + window.location.hostname + ':3004/auth', {
          method: 'POST',
          headers: {
              'Accept': 'application/json',
              'Content-Type': 'application/json'
          },
          body : JSON.stringify(this.state)
        })
        .then(res => {
            if(res.status == 200){
                this.props.setAuthenticate(true);
          
                const cookies = new Cookies();
                cookies.set('token', res.json().token, { path: '/' });
            } else {
                this.setState({ error : true });
            }
        })
        .catch(e => {
          console.log("Failed to authenticate");
          console.log(e);
        })
    }

    render(){   
        let borderColor = "grey";

        if(this.state.error){
            borderColor = "red";
        }

        return(<div style={{marginTop: "20%"}}>
                <div className="row">
                    <div className="col-sm-2"></div>
                    <div className="col-sm-8" style={{textAlign : "center"}}>
                        <h2>Welcome to the Shady Meadows <br /> booking management system</h2>
                    </div>
                    <div className="col-sm-2"></div>
                </div>
                <div className="row">
                    <div className="col-sm-4"></div>
                    <div className="col-sm-4">
                        <p><label htmlFor="username">Username </label><input type="text" id="username" style={{border : "1px solid " + borderColor}} onChange={val => this.setState({username : val.target.value})}/></p> 
                    </div>
                    <div className="col-sm-4"></div>
                </div>
                <div className="row">
                    <div className="col-sm-4"></div>
                    <div className="col-sm-4">
                        <p><label htmlFor="password">Password </label><input type="password" id="password" style={{border : "1px solid " + borderColor}} onChange={val => this.setState({password : val.target.value})}/></p> 
                    </div>
                    <div className="col-sm-4"></div>
                </div>
                <div className="row">
                    <div className="col-sm-7"></div>
                    <div className="col-sm-1">
                        <button type="button" style={{marginLeft : "9px"}} className="btn btn-default" id="doLogin" data-dismiss="modal" onClick={this.doLogin}>Login</button>
                    </div>
                    <div className="col-sm-4"></div>
                </div>
            </div>)
    }

}




