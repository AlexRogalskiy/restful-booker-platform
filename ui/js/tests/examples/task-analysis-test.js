import React from 'react';
import Login from '../../src/js/components/Login.jsx';
import nock from 'nock';

// We first declare our test by using Jests test function and provide it
// with a name and then an anonymous function which will run our test
test('Login component is created', () => {
    // Next we create our component that we want to check. By using
    // enzymes shallow function and passing it the React component
    // we want to create, Enzyme will create a headless DOM and render
    // the React component inside.
    const loginForm = shallow(<Login />);

    // We then compare what the component has done to a previously saved
    // state that lives in the __snapshots__ folder. If anything has changed
    // it will raise an alert.
    expect(loginForm).toMatchSnapshot();
});

// We first declare our test by using Jests test function and provide it
// with a name and then an anonymous function which will run our test
test('Login component sends correct payload', () => {
    // Our Login object is going to send an API request that we want to capture
    // and inspect to make sure the request body matches what we want. We do this
    // by programming a mock using nock with the expected URL and post body and 
    // finally create a fake response for it to reply with
    let authMock = nock('http://localhost:3004')
        .post('/auth/login', {
            "username" : "admin",
            "password" : "password"
        })
        .reply(200, { token: '123ABC' }, () => {
            done();
        });

    // Our component we want to test requires data so we create
    // our object of test data to use later
    const loginState = {
        "error" : false,
        "username" : "admin",
        "password" : "password"
    }

    // Next we create our component that we want to check. By using
    // enzymes shallow function and passing it the React component
    // we want to create, Enzyme will create a headless DOM and render
    // the React component inside.
    const login = shallow(<Login />)

    // Now that we have a rendered React component, we can call it's
    // internal functions. We call setState and pass it our data to 
    // update it's internal state. The component will then process that
    // data and do something with it.
    login.setState(loginState)

    // We next trigger one of the React's custom functions to trigger
    // the login process and send out the HTTP request.
    login.instance().doLogin();

    // Finally, we have to call isDone() on our mock to check whether
    // or not it received the expected request. If it did, it will return
    // true. If it didn't it will return false. We then assert whether 
    // the result is true.
    let didNockAcceptRequest = authMock.isDone();
    expect(didNockAcceptRequest).toBe(true)
})