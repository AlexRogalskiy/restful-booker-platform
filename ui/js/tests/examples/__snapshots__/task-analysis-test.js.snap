// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component is created 1`] = `
<div
  style={
    Object {
      "marginTop": "10%",
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col-sm-2"
    />
    <div
      className="col-sm-8"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <h2>
        Welcome to the Shady Meadows 
        <br />
         booking management system
      </h2>
    </div>
    <div
      className="col-sm-2"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-4"
    />
    <div
      className="col-sm-4"
    >
      <form>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="username"
            onChange={[Function]}
            placeholder="Username"
            style={
              Object {
                "border": "1px solid grey",
              }
            }
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="password"
            onChange={[Function]}
            placeholder="Password"
            style={
              Object {
                "border": "1px solid grey",
              }
            }
            type="password"
          />
        </div>
        <button
          className="btn btn-primary"
          onClick={[Function]}
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
    <div
      className="col-sm-4"
    />
  </div>
</div>
`;
